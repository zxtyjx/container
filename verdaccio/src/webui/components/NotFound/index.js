
import React from 'react';
import PropTypes from 'prop-types';

import classes from './404.scss';

const NotFound = (props) => {
    return (
      <div className={classes.notFound}>
        <h1>Error 404 - {props.pkg}</h1>
        <hr/>
        <p>
          Oops, The package you are trying to access does not exist.
        </p>
      </div>
    );
};

NotFound.propTypes = {
  pkg: PropTypes.string.isRequired
};

export default NotFound;
