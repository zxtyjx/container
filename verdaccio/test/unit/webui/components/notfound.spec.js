/**
 * NotFound component
 */

import React from 'react';
import { shallow } from 'enzyme';
import NotFound from '../../../../src/webui/components/NotFound/index';

console.error = jest.fn();

describe('<NotFound /> component', () => {
  it('should give error for the required fields', () => {
    shallow(<NotFound />);
    expect(console.error).toBeCalled();
  });

  it('should set html from props', () => {
    const props = {
      pkg: 'verdaccio'
    };
    const wrapper = shallow(<NotFound {...props} />);
    expect(wrapper.find('h1').text()).toEqual('Error 404 - verdaccio');
    expect(wrapper.find('p').text()).toEqual(
      'Oops, The package you are trying to access does not exist.'
    );
    expect(wrapper.html()).toMatchSnapshot();
  });
});
