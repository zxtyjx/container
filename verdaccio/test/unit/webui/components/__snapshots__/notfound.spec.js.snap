// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound /> component should set html from props 1`] = `"<div class=\\"notFound\\"><h1>Error 404 - verdaccio</h1><hr/><p>Oops, The package you are trying to access does not exist.</p></div>"`;
